### PR TITLE
5150: Code review fixes

### PR DESCRIPTION
**Pull request by arniloftsson**
#### Link to issue

Please add a link to the issue being addressed by this change.

#### Description

Please include a short description of the suggested change and the reasoning behind the approach you have chosen.

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: modules/ding_seo/ding_seo.module [hunk 1]
```diff
@@ -370,71 +371,55 @@ function ding_seo_xmlsitemap_link_info() {
...
-            'id' => $id,
-            'type' => 'ting_object',
-            'loc' => 'ting/object/' . $object,
-            'language' => 'und',
-            'priority' => 0.5,
-          ]
-        );
-      }
+  foreach ($objects as $id => $object) {
+    // If our id isn't the largest somthings wrong.
```
Comment (holt83) [inline modules/ding_seo/ding_seo.module]: What is wrong and how did we end up here?

Comment (holt83) [reply modules/ding_seo/ding_seo.module]: Also spelling

File: modules/ding_seo/ding_seo.module [hunk 2]
```diff
@@ -370,71 +371,55 @@ function ding_seo_xmlsitemap_link_info() {
...
 function ding_seo_update_sitemap() {
   try {
-    ding_seo_check_sitemap_size();
+    // Check wether to trim any links.
+    $number_of_objects_deleted = ding_seo_check_sitemap_size();
+
     $last_check = variable_get('ding_seo_sitemap_last_check', 0);
-    $objects = ding_seo_get_objects($last_check);
+    // We save the Request time before we do any database operations. So we dont miss
+    // any objects added after we started.
```
Comment (holt83) [inline modules/ding_seo/ding_seo.module:384]: 👍

File: modules/ding_seo/ding_seo.module [hunk 3]
```diff
@@ -370,71 +371,55 @@ function ding_seo_xmlsitemap_link_info() {
...
-      }
+  foreach ($objects as $id => $object) {
+    // If our id isn't the largest somthings wrong.
+    if ($id > $maxid) {
+      $link = [
+        'id' => $id,
+        'type' => 'ting_object',
+        'loc' => 'ting/object/' . $object,
+      ];
+      xmlsitemap_link_save($link);
```
Comment (holt83) [inline modules/ding_seo/ding_seo.module]: Have you done some testing how much longer this will take now?

**Comment by holt83** [comment]
Very fast follow up :) 👍 

Just a couple of questions.

File: modules/ding_seo/ding_seo.module (hunk 3 above)
Comment (arniloftsson) [reply modules/ding_seo/ding_seo.module]: No. But the typical case is about 1000 links per update. Previously a timed a update off 1000 links to about 2.5 seconds so it shouldn't be a problem.

File: modules/ding_seo/ding_seo.module (hunk 1 above)
Comment (arniloftsson) [reply modules/ding_seo/ding_seo.module]: Removed it. Dont need it when we use xmlsitemap_save_link

**Pull request merged**